docs/authentication/stytch/permit-integration.mdx

```diff
@@ -0,0 +1,349 @@
+      </StytchProvider>
+    </>
+  );
+}
+```
+
+### Step 3: Add a StytchLogin Component
+
+Create a StytchLogin component. You can configure which authentication methods you'd like to offer by modifying the products array
+in your config object. Here's an example that includes Email Magic Links and OAuth products:
+
+```javascript
+import { StytchLogin, Products } from "@stytch/vanilla-js";
+
+const Login = () => {
+  const REDIRECT_URL = "SEE_STEP_5";
+
+  const config = {
+    products: [Products.emailMagicLinks, Products.oauth],
+    emailMagicLinksOptions: {
+      loginRedirectURL: REDIRECT_URL,
+      loginExpirationMinutes: 60,
+      signupRedirectURL: REDIRECT_URL,
+      signupExpirationMinutes: 60,
+    },
+    oauthOptions: {
+      providers: [{ type: "google" }],
+      loginRedirectURL: REDIRECT_URL,
+      signupRedirectURL: REDIRECT_URL,
+    },
+  };
+
+  return <StytchLogin config={config} styles={{}} />;
+};
+```
+
+Next, add the StytchLogin component to your application's login page. You'll notice that we check for a logged-in user
+before displaying the StytchLogin component:
+
+```javascript
+import { useStytchUser } from "@stytch/nextjs";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+
+export default function LoginPage() {
+  const { user, isInitialized } = useStytchUser();
+  const router = useRouter();
+
+  useEffect(() => {
+    if (isInitialized && user) {
+      router.replace("/profile");
+    }
+  }, [user, isInitialized, router]);
+
+  return <Login />;
+}
+```
+
+### Step 4: Handle Redirects and Complete the Authentication Flow
+
+Specify the URL that users should be redirected back to by setting the REDIRECT_URL parameter used in your StytchLogin component
+config from Step 3. Be sure to add that URL to the Redirect URLs tab in the Stytch Dashboard.
+
+Implement logic on the page that users are redirected to, retrieving the token value that Stytch adds to the redirect URL query
+parameters. Pass that token to the appropriate Stytch authenticate method:
+
+```javascript
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+import stytch from "stytch";
+
+export default function RedirectPage() {
+  const router = useRouter();
+
+  useEffect(() => {
+    const stytch_token_type = router?.query?.stytch_token_type?.toString();
+    const token = router?.query?.token?.toString();
+
+    if (token && stytch_token_type === "oauth") {
+      stytch.oauth.authenticate(token, {
+        session_duration_minutes: 60,
+      });
+    } else if (token && stytch_token_type === "magic_links") {
+      stytch.magicLinks.authenticate(token, {
+        session_duration_minutes: 60,
+      });
+    }
+  }, [router]);
+
+  return <div>Loading...</div>;
+}
+```
+
+### Step 5: Grant Access to Protected Frontend Content
+
+Upon successful authentication, our frontend JavaScript SDK will automatically populate the session and user objects. Use the
```

Comment:
It's sounds like someone needs to perform successful authentication here, but there is no idea how. Please change the tone.